pkg/tfvars/openstack/openstack.go

```diff
@@ -193,3 +207,96 @@ func getServiceCatalog(cloud string) (*tokens.ServiceCatalog, error) {
 
 	return serviceCatalog, nil
 }
+
+// getImagesCacheDir returns a local path of image cache, where the installer should put the data:
+// <user_cache_dir>/openshift-installer/image_cache
+// If the directory doesn't exist, it will be automatically created.
+func getImagesCacheDir() (string, error) {
+	userCacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return "", err
+	}
+
+	imagesCacheDir := filepath.Join(userCacheDir, "openshift-installer", "image_cache")
+
+	if _, err := os.Stat(imagesCacheDir); os.IsNotExist(err) {
+		err = os.MkdirAll(imagesCacheDir, 0755)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return imagesCacheDir, nil
+}
+
+// downloadOpenStackImageFile obtains a file from a given URL, puts it in the cache
+// and returns the local file path.
+func downloadOpenStackImageFile(baseURL string) (string, error) {
```

Comment:
also it looks like https://github.com/openshift/installer/blob/6669a800bc89997dbc806443f32dcf32286fd93f/pkg/tfvars/libvirt/cache.go and this could share a lot of the code...

maybe `pkg/tfvars/internal` can hold this to prevent expsoing this too much
https://golang.org/doc/go1.4#internalpackages